parsers/cxx/cxx_parser_block.c

```diff
@@ -582,3 +582,19 @@ bool cxxParserParseBlock(bool bExpectClosingBracket)
 	CXX_DEBUG_LEAVE_TEXT("WARNING: Not reached");
 	return true;
 }
+
+//
+// This is the toplevel scanning function. It's a forward-only scanner that keeps
+// accumulating tokens in the chain until either a characteristic token is found
+// or the statement ends. When a characteristic token is found it usually enters
+// a specialized scanning routine (e.g for classes, namespaces, structs...).
+// When the statement ends without finding any characteristic token the chain
+// is passed to an analysis routine which does a second scan pass.
+//
+bool cxxParserParseBlock(bool bExpectClosingBracket)
+{
+	cppSetExternalParserState(cppGetExternalParserState() + 1);
+	bool bRet = cxxParserParseBlockInternal(bExpectClosingBracket);
+	cppSetExternalParserState(cppGetExternalParserState() - 1);
```

Comment:
Does Cxx parser uses the value returned from cppGetExternalParserState()?
If not, how about introducing cppPushExternalParserState/cppPopExternalParserState instead of Get/Set?
So you don't have to write +/- 1 for each invocation.